test/unit/scratchOrgInfoApiTest.ts

```diff
@@ -0,0 +1,539 @@
+  SignupEmail: 'sfdx-cli@salesforce.com',
+  SignupUsername: 'sfdx-cli',
+  Username: 'sfdx-cli',
+  SignupInstance: 'http://salesforce.com',
+};
+
+describe('requestScratchOrgCreation', () => {
+  const sandbox = sinon.createSandbox();
+  const connectionStub = sinon.createStubInstance(Connection);
+
+  beforeEach(() => {
+    stubMethod(sandbox, Org, 'create').resolves(Org.prototype);
+    connectionStub.sobject.withArgs('ScratchOrgInfo').returns({
+      create: sinon.stub().resolves(true),
+    } as unknown as SObject<unknown>);
+    stubMethod(sandbox, Org.prototype, 'getConnection').returns(connectionStub);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('requestScratchOrgCreation', async () => {
+    const error = new Error('MyError');
+    error.name = 'NamedOrgNotFound';
+    stubMethod(sandbox, AuthInfo, 'create').rejects(error);
+    const hubOrg = new Org({});
+    const settings = new SettingsGenerator();
+    const result = await requestScratchOrgCreation(hubOrg, TEMPLATE_SCRATCH_ORG_INFO, settings);
+    expect(result).to.be.true;
+  });
+
+  it('requestScratchOrgCreation no username field in scratchOrgRequest', async () => {
+    const scratchOrgRequest = Object.assign({}, TEMPLATE_SCRATCH_ORG_INFO);
+    delete scratchOrgRequest.Username;
+    const hubOrg = new Org({});
+    const settings = new SettingsGenerator();
+    const result = await requestScratchOrgCreation(hubOrg, scratchOrgRequest, settings);
+    expect(result).to.be.true;
+  });
+
+  it('requestScratchOrgCreation no username field in scratchOrgRequest is empty', async () => {
+    const scratchOrgRequest = Object.assign({}, TEMPLATE_SCRATCH_ORG_INFO);
+    scratchOrgRequest.Username = undefined;
+    const hubOrg = new Org({});
+    const settings = new SettingsGenerator();
+    const result = await requestScratchOrgCreation(hubOrg, scratchOrgRequest, settings);
+    expect(result).to.be.true;
+  });
+
+  it('requestScratchOrgCreation user exists', async () => {
+    stubMethod(sandbox, AuthInfo, 'create').resolves();
+    const hubOrg = new Org({});
+    const settings = new SettingsGenerator();
+    try {
+      await shouldThrow(requestScratchOrgCreation(hubOrg, TEMPLATE_SCRATCH_ORG_INFO, settings));
+    } catch (error) {
+      expect(error).to.exist;
+      expect(error.message).to.include(errorCodesMessages.getMessage('C-1007'));
+    }
+  });
+
+  it('requestScratchOrgCreation JsForce Error NamedOrgNotFound', async () => {
+    const err = new Error('MyError');
+    err.name = 'NamedOrgNotFound';
+    stubMethod(sandbox, AuthInfo, 'create').rejects(err);
+    connectionStub.sobject.withArgs('ScratchOrgInfo').returns({
+      create: sinon.stub().rejects(),
+    } as unknown as SObject<unknown>);
+    const hubOrg = new Org({});
+    const settings = new SettingsGenerator();
+    try {
+      await shouldThrow(requestScratchOrgCreation(hubOrg, TEMPLATE_SCRATCH_ORG_INFO, settings));
+    } catch (error) {
+      expect(error).to.exist;
```

Comment:
NIT: do we need to assert any values on the error?

Reply:
YES! this is interesting because right below there are two a very similar but completely different type of errors thrown by `jsforce` and we need to test them all so it makes sense to test for the `error.message`